include/gmock-win32.h

```diff
@@ -50,6 +50,22 @@ namespace detail {
         return ref_proxy< Reference >{ std::forward< Reference >(r) };
     }
 
+    template< typename Derived >
+    struct mock_module_base
+    {
+        static Derived& instance()
+        {
+            static ::testing::NiceMock< Derived > obj;
+            return obj;
+        }
+
+        static void** ppOldFn()
+        {
+            static void* oldFn_ = nullptr;
+            return &oldFn_;
+        }
+    };
+
```

Comment:
When using inheritance, there are not many macro edits.

Reply:
I agree, maybe we need to try to reduce the code by using a base structure (but it is not easy when there are many arguments)